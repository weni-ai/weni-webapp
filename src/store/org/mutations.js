export default {
  setCurrentOrg(state, organization) {
    state.currentOrg = organization;
  },

  ORG_CREATE_REQUEST: (state) => (state.loadingCreateOrg = true),
  ORG_CREATE_SUCCESS: (state, org) => {
    state.currentOrg = org;
    state.loadingCreateOrg = false;
  },
  ORG_CREATE_ERROR: (state, createOrgError) => {
    state.errorCreateOrg = createOrgError;
    state.loadingCreateOrg = false;
  },
};
