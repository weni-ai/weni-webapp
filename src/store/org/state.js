export default {
  currentOrgId: null,
  currentOrg: null,
  loadingCreateOrg: false,
  errorCreateOrg: false,
};
