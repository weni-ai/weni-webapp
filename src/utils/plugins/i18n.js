import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import es from '../../locales/es.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  'pt-br': pt_br,
  'pt-BR': pt_br,
  en: en,
  es: es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
