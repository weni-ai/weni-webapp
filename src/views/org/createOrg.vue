<template>
  <container class="weni-create-org" :center="current === 3">
    <indicator
      class="weni-create-org__indicator"
      :steps="steps.length"
      :current="current + 1"
      :names="steps"
    />

    <div v-show="current === 0" class="weni-create-org__section">
      <div class="title">
        {{ $t('orgs.create.organization_title') }}
      </div>

      <unnnic-input
        class="weni-create-org__name-input"
        v-model="orgName"
        :label="$t('orgs.create.org_name')"
        :placeholder="$t('orgs.create.org_name_placeholder')"
      />
      <unnnic-input
        v-model="orgDescription"
        :label="$t('orgs.create.org_description')"
        :placeholder="$t('orgs.create.org_description_placeholder')"
      />
      <div class="weni-create-org__group weni-create-org__group__buttons">
        <unnnic-button @click="back" type="terciary">
          {{ $t('orgs.create.back') }}
        </unnnic-button>
        <unnnic-button
          :disabled="!canProgress"
          type="secondary"
          @click="
            setBillingOrgStep({ name: orgName, description: orgDescription })
          "
        >
          {{ $t('orgs.create.next') }}
        </unnnic-button>
      </div>
    </div>
    <div v-show="current === 1" class="weni-create-org__section">
      <div class="title">
        {{ $t('orgs.create.title') }}
      </div>

      <user-management
        :label-role="$t('orgs.create.permission')"
        :label-email="$t('orgs.create.org_user_email')"
        do-not-fetch
        cannot-delete-my-user
        tooltip-side-icon-right="right"
        :users="users"
        @users="users = $event"
        :changes="userChanges"
        @changes="userChanges = $event"
        :style="{
          display: 'flex',
          flexDirection: 'column',
        }"
        :already-added-text="$t('orgs.users.already_added')"
      ></user-management>

      <div class="weni-create-org__group weni-create-org__group__buttons">
        <unnnic-button type="terciary" @click="backBilling">
          {{ $t('orgs.create.back') }}
        </unnnic-button>
        <unnnic-button type="secondary" @click="onProceedPermissions()">
          {{ $t('orgs.create.next') }}
        </unnnic-button>
      </div>
    </div>
    <div v-show="current === 2" class="weni-create-org__section">
      <div class="title">
        {{ $t('orgs.create.project_title') }}
      </div>

      <unnnic-input
        v-model="projectName"
        :label="$t('orgs.create.project_name')"
        :placeholder="$t('orgs.create.project_name_placeholder')"
      />
      <unnnic-select
        v-model="dateFormat"
        :label="$t('orgs.create.date_format')"
      >
        <option value="D">DD-MM-YYYY</option>
        <option value="M">MM-DD-YYYY</option>
      </unnnic-select>

      <unnnic-select
        v-model="timeZone"
        :label="$t('orgs.create.time_zone')"
        search
        :search-placeholder="$t('orgs.create.timezone_search_placeholder')"
      >
        <option
          v-for="timezone in timezones"
          :key="timezone.zoneName"
          :value="timezone.zoneName"
        >
          {{ timezone }}
        </option>
      </unnnic-select>

      <div class="weni-create-org__group weni-create-org__group__buttons">
        <unnnic-button
          type="terciary"
          :disabled="creationFreeLoading"
          @click="backBilling"
        >
          {{ $t('orgs.create.back') }}
        </unnnic-button>
        <unnnic-button
          :disabled="!canProgress"
          :loading="creationFreeLoading"
          type="secondary"
          @click="tempCreateOrg"
        >
          <!--
            Temporary: replace the above @click for the below one
            @click="
              setBillingProjectStep({ name: projectName, dateFormat, timeZone })
            "
          -->
          {{ $t('orgs.create.done') }}
        </unnnic-button>
      </div>
    </div>
    <div v-if="current === 3">
      <!-- <BillingCreateOrg /> -->
    </div>
    <div v-show="current === 3" class="weni-create-org__section">
      <h1>
        {{ $t('orgs.create.finish_text') }}
        <emoji name="Winking Face" />
      </h1>

      <p class="weni-create-org__error" v-if="error">
        {{ $t('orgs.create.save_error') }}
      </p>
      <div class="weni-create-org__group weni-create-org__group__buttons">
        <unnnic-button @click="viewProjects" type="terciary">{{
          $t('projects.create.view_projects')
        }}</unnnic-button>
        <unnnic-button @click="onFinish" type="secondary">
          {{ $t('orgs.create.go_to_org') }}
        </unnnic-button>
      </div>
    </div>
  </container>
</template>

<script>
import Indicator from '../../components/orgs/indicator';
import UserManagement from '../../components/orgs/UserManagement.vue';
import Emoji from '../../components/Emoji.vue';
import timezones from '../projects/timezone';
import container from '../projects/container';
/* Temporary: remove the comment
import BillingCreateOrg from '@/views/billing/createOrg.vue';
*/
import _ from 'lodash';
import orgs from '../../api/orgs';

import { unnnicCallAlert } from '@weni/unnnic-system';
import { mapActions, mapState } from 'vuex';

export default {
  name: 'CreateOrg',
  components: {
    Indicator,
    UserManagement,
    Emoji,
    container,
    /* Temporary: remove the comment
    BillingCreateOrg,
    */
  },

  mixins: [timezones],

  data() {
    return {
      loading: false,
      org: null,
      project: null,
      error: null,
      orgError: null,
      orgName: null,
      orgDescription: null,
      projectName: null,
      dateFormat: 'D',
      timeZone: 'America/Argentina/Buenos_Aires',
      users: [],
      userChanges: {},
    };
  },
  computed: {
    ...mapState({
      current: (state) => state.BillingSteps.current,
      // users: (state) => state.BillingSteps.users,
      organizationCreationError: (state) => state.Org.errorCreateOrg,
      projectCreationError: (state) => state.Project.errorCreateProject,
      creationFreeLoading: (state) =>
        state.Org.loadingCreateOrg || state.Project.loadingCreateProject,
    }),
    steps() {
      return ['organization', 'members', 'project'].map((name) =>
        this.$t(`orgs.create.${name}`),
      );
    },
    canProgress() {
      console.log(this.current);
      if (this.current === 0) {
        return [this.orgName, this.orgDescription].every(
          (field) => field && field.length > 0,
        );
      }
      if (this.current === 1) return true;
      if (this.current === 2) {
        return [this.projectName, this.dateFormat, this.timeZone].every(
          (field) => field && field.length > 0,
        );
      }
      return true;
    },
  },

  created() {
    const ROLE_ADMIN = '3';
    const user = this.$store.state.Account.profile;

    this.users = [
      {
        id: user.id,
        uuid: null,
        name: [user.first_name, user.last_name]
          .filter((name) => name)
          .join(' '),
        email: user.email,
        photo: user.photo,
        role: ROLE_ADMIN,
        username: user.username,
      },
    ];

    this.$store.state.BillingSteps.current = 0;
  },

  methods: {
    ...mapActions([
      'createOrg',
      'changeAuthorization',
      'createProject',
      'setCurrentOrg',
      'setCurrentProject',
      'openModal',
      'setBillingOrgStep',
      'setBillingMembersStep',
      'setBillingProjectStep',
      'backBilling',
    ]),

    /* Temporary: remove tempCreateOrg method */
    async tempCreateOrg() {
      this.setBillingProjectStep({
        name: this.projectName,
        dateFormat: this.dateFormat,
        timeZone: this.timeZone,
      });

      await this.createOrg('free');
      await this.createProject();

      if (this.organizationCreationError || this.projectCreationError) {
        this.openModal({
          type: 'alert',
          data: {
            persistent: true,
            icon: 'alert-circle-1',
            scheme: 'feedback-yellow',
            title: this.$t('alerts.server_problem.title'),
            description: this.$t('alerts.server_problem.description'),
          },
        });
      } else {
        this.$store.state.BillingSteps.current++;
      }
    },

    openServerErrorAlertModal({
      title = this.$t('alerts.server_problem.title'),
      description = this.$t('alerts.server_problem.description'),
    } = {}) {
      this.openModal({
        type: 'alert',
        data: {
          icon: 'alert-circle-1',
          scheme: 'feedback-yellow',
          title,
          description,
        },
      });
    },

    back() {
      if (
        !this.orgName &&
        !this.orgDescription &&
        !this.projectName &&
        this.users.length === 1
      ) {
        this.backBilling();

        return this.$router.push('/orgs');
      }
      this.openModal({
        type: 'confirm',
        data: {
          persistent: true,
          icon: 'alert-circle-1',
          scheme: 'feedback-yellow',
          title: this.$t('orgs.create.exit_org_creation'),
          description: this.$t('orgs.create.exit_org_creation_description'),
          cancelText: this.$t('cancel'),
          confirmText: this.$t('orgs.create.no_permission_confirm'),
          onConfirm: (justClose) => {
            justClose();
            this.backBilling();
            this.$router.push('/orgs');
          },
        },
      });
    },
    onProceedPermissions() {
      if (this.users.length === 1) {
        this.openModal({
          type: 'confirm',
          data: {
            persistent: true,
            icon: 'alert-circle-1',
            scheme: 'feedback-yellow',
            title: this.$t('orgs.create.no_permission_title'),
            description: this.$t('orgs.create.no_permission'),
            cancelText: this.$t('cancel'),
            confirmText: this.$t('orgs.create.no_permission_confirm'),
            onConfirm: (justClose) => {
              justClose();
              this.setBillingMembersStep({
                users: this.users,
                userChanges: this.userChanges,
              });
            },
          },
        });
      } else {
        this.setBillingMembersStep({
          users: this.users,
          userChanges: this.userChanges,
        });
      }
    },

    async onCreateOrg() {
      try {
        const response = await this.createOrg({
          name: this.orgName,
          description: this.orgDescription,
        });
        this.org = response.data;
      } catch (e) {
        console.log(e);
        this.orgError = e;
      }
    },
    async onMakeChanges() {
      var changes = Object.values(this.userChanges).map(async (change) => {
        try {
          const organizationUuid = _.get(this.org, 'uuid');

          await orgs.createRequestPermission({
            organization: organizationUuid,
            email: change.email,
            role: change.role,
          });
        } catch (e) {
          console.log(e);
          this.error = true;
        }
      });
      const createProject = async () => {
        try {
          const response = await this.createProject({
            orgId: this.org.uuid,
            name: this.projectName,
            dateFormat: this.dateFormat,
            timezone: this.timeZone,
          });

          this.project = response.data;
        } catch (e) {
          this.setCurrentOrg(this.org);

          this.$router.push({
            name: 'projects',
            params: {
              orgUuid: this.org.uuid,
            },
          });

          unnnicCallAlert({
            props: {
              icon: 'alert-circle-1-1',
              scheme: 'feedback-yellow',
              text: this.$t('projects.create.error'),
              title: '',
              position: 'bottom-right',
              closeText: this.$t('close'),
            },
            seconds: 3,
          });
        }
      };
      changes = [...changes, createProject()];
      await Promise.all(changes);
    },
    async onSubmit() {
      this.loading = true;
      await this.onCreateOrg();

      if (this.orgError) {
        this.openServerErrorAlertModal();
        this.orgError = null;
      } else {
        await this.onMakeChanges();
        this.current = this.current + 1;
      }
      this.loading = false;
    },

    viewProjects() {
      this.$router.replace({
        name: 'projects',
        params: {
          orgUuid: this.$store.state.Org.currentOrg.uuid,
        },
      });
    },

    onFinish() {
      this.$router.replace({
        name: 'home',
        params: {
          projectUuid: this.$store.state.Project.currentProject.uuid,
        },
      });
      this.$root.$emit('set-sidebar-expanded');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';

.title {
  color: $unnnic-color-neutral-darkest;
  font-family: $unnnic-font-family-primary;
  font-weight: $unnnic-font-weight-regular;
  font-size: $unnnic-font-size-title-md;
  line-height: $unnnic-font-size-title-md + $unnnic-line-height-md;
  text-align: center;
  margin-bottom: $unnnic-spacing-stack-md;
}

.weni-create-org__name-input {
  margin-bottom: $unnnic-spacing-stack-md;
}
</style>

<style lang="scss">
@import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';
.weni-create-org {
  h1 {
    font-size: $unnnic-font-size-title-md;
    font-family: $unnnic-font-family-primary;
    font-weight: $unnnic-font-weight-regular;
    margin: 0 0 $unnnic-spacing-stack-md 0;
    text-align: center;
  }

  &__error {
    font-family: $unnnic-font-family-secondary;
    text-align: center;
  }

  &__indicator {
    margin: 0 auto;
    margin-bottom: 22px + 40px;
    max-width: 50%;
  }

  &__group {
    display: flex;
    > *:not(:last-child) {
      margin-right: $unnnic-spacing-stack-sm;
    }

    &__buttons {
      margin-top: $unnnic-spacing-stack-md;
      > * {
        flex: 1;
      }
    }
  }

  &__section {
    width: 100%;
  }
}
</style>
