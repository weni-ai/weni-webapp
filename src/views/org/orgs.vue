<template>
  <div class="weni-orgs">
    <div v-show="organizationsStatus !== 'loading'" class="unnnic-grid-lg">
      <div class="weni-orgs__left unnnic-grid-span-5">
        <div :class="['box', 'aux-blue', 'weni-orgs__left__icon']">
          <unnnic-icon-svg icon="building-2-1" size="xl" scheme="aux-blue" />
        </div>

        <h1>{{ $t('orgs.orgs') }}</h1>

        <template v-if="error">
          <p>{{ $t('orgs.error_on_loading_orgs') }}</p>

          <unnnic-button
            type="secondary"
            icon-left="button-refresh-arrow-1"
            @click="tryAgain()"
            :disabled="organizationsStatus === 'loading'"
          >
            {{ $t('try_again') }}
          </unnnic-button>
        </template>

        <template v-else>
          <p>
            {{
              $t(
                organizationsStatus === 'empty'
                  ? 'orgs.orgs_description_empty'
                  : 'orgs.orgs_description',
              )
            }}
          </p>

          <router-link to="/orgs/create">
            <unnnic-button type="secondary" icon-left="add-1">
              {{ $t('orgs.add_org') }}
            </unnnic-button>
          </router-link>
        </template>
      </div>
      <div class="unnnic-grid-span-2" />
      <div class="unnnic-grid-span-5 weni-orgs__right">
        <img
          v-if="organizationsStatus === 'empty'"
          src="@/assets/empty-orgs.svg"
          width="100%"
        />

        <div v-else class="weni-orgs__list">
          <org-list
            class="list-container"
            ref="orgList"
            @status="organizationsStatus = $event"
          />
        </div>
      </div>
    </div>
    <skeleton-loading v-show="organizationsStatus === 'loading'" />
  </div>
</template>

<script>
import OrgList from '../../components/orgs/orgList.vue';
import SkeletonLoading from '../loadings/orgs.vue';
import { mapActions } from 'vuex';

export default {
  name: 'Orgs',
  components: {
    OrgList,
    SkeletonLoading,
  },

  data() {
    return {
      error: false,
      organizationsStatus: '',
    };
  },

  mounted() {
    this.clearCurrentOrg();
    this.clearCurrentProject();
  },

  watch: {
    organizationsStatus(status) {
      if (status === 'error') {
        this.error = true;
      } else if (status === 'loaded' || status === 'empty') {
        this.error = false;
      }
    },
  },

  methods: {
    ...mapActions(['clearCurrentOrg', 'clearCurrentProject']),

    tryAgain() {
      this.$refs.orgList.reloadOrganizations();
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';

.weni-orgs {
  display: flex;
  flex-direction: column;

  .box {
    border-radius: $unnnic-border-radius-sm;
    padding: $unnnic-spacing-inset-nano;

    &.aux-blue {
      background-color: rgba(
        $unnnic-color-aux-blue,
        $unnnic-opacity-level-extra-light
      );
    }
  }

  a {
    text-decoration: none;
  }

  .unnnic-grid-lg {
    flex: 1;
    padding: 0;
    box-sizing: border-box;
    padding: 0 12.88%;
    padding-bottom: $unnnic-spacing-stack-xl - ($unnnic-border-width-thick * 2);
    border-bottom: $unnnic-border-width-thick * 2 solid $unnnic-color-brand-weni;
  }

  &__right {
    display: flex;
    flex-direction: column;
    justify-content: center;

    &::-webkit-scrollbar {
      display: none;
    }
  }

  &__list {
    overflow-y: scroll;
    -ms-overflow-style: none; /* IE and Edge */
    scrollbar-width: none;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    max-height: 100%;
    height: 100%;

    > * {
      margin-bottom: $unnnic-spacing-stack-xs;
    }

    .list-container {
      margin: 0;
      max-height: 0;
    }

    flex: 1;
    overflow: overlay;
    min-height: 4rem;

    $scroll-size: $unnnic-inline-nano;
    padding-right: calc(#{$unnnic-inline-xs} + #{$scroll-size});
    width: 100%;

    &::-webkit-scrollbar {
      width: $scroll-size;
    }

    &::-webkit-scrollbar-thumb {
      background: $unnnic-color-neutral-clean;
      border-radius: $unnnic-border-radius-pill;
    }

    &::-webkit-scrollbar-track {
      background: $unnnic-color-neutral-soft;
      border-radius: $unnnic-border-radius-pill;
    }
  }

  &__left {
    font-family: $unnnic-font-family-primary;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    height: 100%;

    &__icon {
      margin: 0 0 $unnnic-spacing-stack-md 0;
    }

    h1 {
      font-size: $unnnic-font-size-title-lg;
      font-weight: $unnnic-font-weight-regular;
      margin: 0 0 $unnnic-spacing-stack-xs 0;
    }

    p {
      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-lg;
      margin: 0 0 $unnnic-spacing-stack-md 0;
      color: $unnnic-color-neutral-dark;
    }
  }
}
</style>
