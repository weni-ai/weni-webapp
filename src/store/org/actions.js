import orgs from '../../api/orgs';

export default {
  getOrgs(store, { page = 1, limit = 20 }) {
    const offset = limit * (page - 1);
    return orgs.list(offset, limit);
  },

  async createOrg(
    {
      commit,
      rootState: {
        BillingSteps: { org },
      },
    },
    orgType,
  ) {
    commit('ORG_CREATE_REQUEST');
    try {
      const response = await orgs.createOrg(org.name, org.description, orgType);
      commit('ORG_CREATE_SUCCESS', response.data);
    } catch (e) {
      commit('ORG_CREATE_ERROR', e);
      commit('OPEN_MODAL', {});
    }
  },

  getOrg(store, { uuid }) {
    return orgs.getOrg({ uuid });
  },

  editOrg(store, { uuid, name, description }) {
    return orgs.editOrg(uuid, name, description);
  },

  deleteOrg(store, { uuid }) {
    return orgs.deleteOrg(uuid);
  },

  getMembers(store, { uuid, page = 1, limit = 20, search }) {
    const offset = limit * (page - 1);
    return orgs.getMembers(uuid, offset, limit, search);
  },

  createRequestPermission(store, data) {
    return orgs.createRequestPermission(data);
  },

  addAuthorization(store, { orgId, username, role }) {
    return orgs.addAuthorization(orgId, username, role);
  },

  removeAuthorization(store, { orgId, username }) {
    return orgs.removeAuthorization(orgId, username);
  },

  changeAuthorization(store, { orgId, username, role }) {
    return orgs.changeAuthorization(orgId, username, role);
  },

  leaveOrg(store, { orgId, id }) {
    return orgs.leaveOrg(orgId, id);
  },

  setCurrentOrg(
    { commit },
    {
      name,
      uuid,
      inteligence_organization,
      authorization,
      organization_billing,
      extra_integration,
    } = {},
  ) {
    commit('setCurrentOrg', {
      name,
      uuid,
      inteligence_organization,
      authorization,
      organization_billing,
      extra_integration,
    });
  },

  clearCurrentOrg({ commit }) {
    commit('setCurrentOrg', null);
  },
};
