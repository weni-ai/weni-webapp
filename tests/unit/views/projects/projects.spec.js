import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Router from 'vue-router';
import Projects from '@/views/projects/projects.vue';
import i18n from '@/utils/plugins/i18n';
import org from '../../../__mocks__/org';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Router);

const router = new Router();

jest.mock('@/services/SecurityService.js', () => {});
jest.mock('@/api/request.js', () => {});

describe('Projects.vue', () => {
  let wrapper;
  let store;
  let actions;
  let getters;

  beforeEach(() => {
    getters = {
      currentOrg: () => {
        return org;
      },
    };
    actions = {
      setCurrentProject: jest.fn(),
    };

    store = new Vuex.Store({
      getters,
      actions,
    });

    wrapper = shallowMount(Projects, {
      localVue,
      i18n,
      store,
      router,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        RouterLink: RouterLinkStub,
        ProjectList: true,
        ProjectLoading: true,
        RightSideBar: true,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('openManageMembers()', () => {
    it('Set isMemberManagementBarOpen true', () => {
      wrapper.vm.openManageMembers();
      expect(wrapper.vm.isMemberManagementBarOpen).toBe(true);
    });
  });

  describe('openViewMembers()', () => {
    it('Set isMemberViewerBarOpen true', () => {
      wrapper.vm.openViewMembers();
      expect(wrapper.vm.isMemberViewerBarOpen).toBe(true);
    });
  });

  describe('selectProject()', () => {
    it('If there are route in params', () => {
      wrapper.vm.selectProject({ uuid: '' }, '/orgs');
      expect(wrapper.vm.$router.history.current.path).toBe('/orgs');
      expect(wrapper.emitted('set-sidebar-expanded'));
    });
  });

  describe('loadingProject()', () => {
    it('Test loading projects', () => {
      wrapper.vm.loadingProject(true);
      expect(wrapper.vm.loadingProjects).toBe(true);
    });
  });

  describe('loadingProject()', () => {
    it('Test loading projects', () => {
      wrapper.vm.loadingProject(true);
      expect(wrapper.vm.loadingProjects).toBe(true);
    });
  });
});
