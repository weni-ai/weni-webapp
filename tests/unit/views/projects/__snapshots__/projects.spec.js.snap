// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects.vue should be rendered properly 1`] = `
<div class="weni-projects">
  <div class="container">
    <div class="unnnic-grid-span-12 content">
      <div class="header">
        <div class="unnnic-grid-lg">
          <div class="unnnic-grid-span-6 title-container">
            <div class="title">
              some specific text
            </div>
          </div>
          <div class="unnnic-grid-span-6 subtitle-container">
            <div class="subtitle">
              some specific text
            </div>
          </div>
          <div class="
              unnnic-grid-span-3
              view-or-menage-organization-members-button-container
            ">
            <unnnic-button-stub size="large" type="secondary" iconleft="view-1-1" class="button">
              some specific text
            </unnnic-button-stub>
          </div>
          <div class="unnnic-grid-span-3 change-organization-button-container"><a>
              <unnnic-button-stub size="large" type="secondary" iconleft="button-refresh-arrows-1" class="button">
                some specific text
              </unnnic-button-stub>
            </a></div>
        </div>
      </div>
      <div class="line"></div>
      <div class="order"><span class="label">
          some specific text
        </span>
        <unnnic-radio-stub value="lastAccess" globalvalue="lastAccess" size="md" class="radio">
          some specific text
        </unnnic-radio-stub>
        <unnnic-radio-stub value="alphabetical" globalvalue="lastAccess" size="md" class="radio">
          some specific text
        </unnnic-radio-stub>
        <unnnic-radio-stub value="newer" globalvalue="lastAccess" size="md" class="radio">
          some specific text
        </unnnic-radio-stub>
        <unnnic-radio-stub value="older" globalvalue="lastAccess" size="md" class="radio">
          some specific text
        </unnnic-radio-stub>
      </div>
      <div class="projects-list-container">
        <div class="projects-list" style="padding-right: 15px;">
          <projectlist-stub org="123" order="lastAccess"></projectlist-stub>
        </div>
      </div>
    </div>
    <div class="unnnic-grid-span-12 content" style="display: none;">
      <projectloading-stub></projectloading-stub>
    </div>
  </div>
  <rightsidebar-stub type="view-members" props="[object Object]"></rightsidebar-stub>
  <rightsidebar-stub type="manage-members" props="[object Object]"></rightsidebar-stub>
</div>
`;
